src/execution/mod.rs

```diff
@@ -40,11 +39,65 @@ use {
     tokio::sync::Mutex, tonic::transport::Channel,
 };
 
-/// Structure for executing queries either locally or remotely (via FlightSQL)
+/// Provides all core execution functionality for execution queries from either a local
+/// `SessionContext` or a remote `FlightSQL` service
+pub struct AppExecution {
+    context: ExecutionContext,
+    #[cfg(feature = "flightsql")]
+    flightsql_client: Mutex<Option<FlightSqlServiceClient<Channel>>>,
+}
```

Comment:
`AppExecution` is the execution component used by all dft client apps (cli, tui).  `ExecutionContext` is specifically for local datafusion contexts, including the flightsql server. the flightsql client had to be extracted as it wasnt `Clone` so it couldnt be used with the flightsql server.